### PR TITLE
update requirements files

### PR DESCRIPTION
**Pull request by blentz**
Updating our version pins to use current versions of most dependencies.

This includes updating the Azure dependencies across a breaking API change. Updating those API references is also included in this PR. This matches the same updates we did recently in Nise.

Functional Testing:

Creating an Azure provider:
```
[2020-02-12 18:35:41,331] INFO None GET: /api/cost-management/v1/reports/aws/costs/ -- ACCOUNT: 10001 USER: user_dev ORG_ADMIN: True REQ_ID: None
[2020-02-12 18:35:41,366] DEBUG None API: /api/cost-management/v1/reports/aws/costs/ USER: user_dev
[2020-02-12 18:35:41,375] DEBUG None Query Parameters: {'parameters': {'compute_count': False,
                'filter': OrderedDict([('time_scope_value', '-10'),
                                       ('time_scope_units', 'day'),
                                       ('resolution', 'daily')]),
                'group_by': OrderedDict(),
                'order_by': OrderedDict()},
 'query_handler': <class 'api.report.aws.query_handler.AWSReportQueryHandler'>,
 'report_type': 'costs',
 'request': <rest_framework.request.Request object at 0x7fbe841b1470>,
 'serializer': <class 'api.report.aws.serializers.QueryParamSerializer'>,
 'tag_handler': [<class 'reporting.provider.aws.models.AWSTagsSummary'>],
 'tag_keys': []}
[2020-02-12 18:35:41,379] DEBUG None Query Params: {'parameters': {'compute_count': False,
                'filter': OrderedDict([('time_scope_value', '-10'),
                                       ('time_scope_units', 'day'),
                                       ('resolution', 'daily')]),
                'group_by': OrderedDict(),
                'order_by': OrderedDict()},
 'query_handler': <class 'api.report.aws.query_handler.AWSReportQueryHandler'>,
 'report_type': 'costs',
 'request': <rest_framework.request.Request object at 0x7fbe841b1470>,
 'serializer': <class 'api.report.aws.serializers.QueryParamSerializer'>,
 'tag_handler': [<class 'reporting.provider.aws.models.AWSTagsSummary'>],
 'tag_keys': []}
[2020-02-12 18:35:41,380] DEBUG None Query Params: {'parameters': {'compute_count': False,
                'filter': OrderedDict([('time_scope_value', '-10'),
                                       ('time_scope_units', 'day'),
                                       ('resolution', 'daily')]),
                'group_by': OrderedDict(),
                'order_by': OrderedDict()},
 'query_handler': <class 'api.report.aws.query_handler.AWSReportQueryHandler'>,
 'report_type': 'costs',
 'request': <rest_framework.request.Request object at 0x7fbe841b1470>,
 'serializer': <class 'api.report.aws.serializers.QueryParamSerializer'>,
 'tag_handler': [<class 'reporting.provider.aws.models.AWSTagsSummary'>],
 'tag_keys': []}
[2020-02-12 18:35:41,381] DEBUG None _get_search_filter: (AND: ('usage_start__date__gte', datetime.datetime(2020, 2, 3, 0, 0, tzinfo=<UTC>)), ('usage_end__date__lte', datetime.datetime(2020, 2, 12, 18, 0, tzinfo=<UTC>)))
[2020-02-12 18:35:41,381] DEBUG None _get_filter: (AND: ('usage_start__date__gte', datetime.datetime(2020, 2, 3, 0, 0, tzinfo=<UTC>)), ('usage_end__date__lte', datetime.datetime(2020, 2, 12, 18, 0, tzinfo=<UTC>)))
[2020-02-12 18:35:41,397] DEBUG None DATA: {'compute_count': False, 'filter': OrderedDict([('time_scope_value', '-10'), ('time_scope_units', 'day'), ('resolution', 'daily')]), 'group_by': OrderedDict(), 'order_by': OrderedDict(), 'data': [{'date': '2020-02-03', 'values': []}, {'date': '2020-02-04', 'values': []}, {'date': '2020-02-05', 'values': []}, {'date': '2020-02-06', 'values': []}, {'date': '2020-02-07', 'values': []}, {'date': '2020-02-08', 'values': []}, {'date': '2020-02-09', 'values': []}, {'date': '2020-02-10', 'values': []}, {'date': '2020-02-11', 'values': []}, {'date': '2020-02-12', 'values': []}], 'total': {'cost': {'value': 0, 'units': 'USD'}, 'infrastructure_cost': {'value': 0, 'units': 'USD'}, 'derived_cost': {'value': 0, 'units': 'USD'}, 'markup_cost': {'value': 0, 'units': 'USD'}}}
[2020-02-12 18:35:41,401] INFO None GET: /api/cost-management/v1/reports/aws/costs/ 200 -- ACCOUNT: 10001 USER: user_dev ORG_ADMIN: True REQ_ID: None

```

Downloading a report:
```
User information: uid=1000890000 euid=1000890000 gid=0 egid=0

  uid=uid, euid=euid, gid=gid, egid=egid,
 
 -------------- celery@koku-worker-0 v4.4.0 (cliffs)
--- ***** ----- 
-- ******* ---- Linux-4.18.0-147.3.1.el8_1.x86_64-x86_64-with-redhat-7.7-Maipo 2020-02-12 18:18:09
- *** --- * --- 
- ** ---------- [config]
- ** ---------- .> app:         koku:0x7fa94654f470
- ** ---------- .> transport:   amqp://guest:**@rabbitmq:5672//
- ** ---------- .> results:     disabled://
- *** --- * --- .> concurrency: 1 (prefork)
-- ******* ---- .> task events: ON
--- ***** ----- 
 -------------- [queues]
                .> celery           exchange=celery(direct) key=celery
                .> customer_data_sync exchange=customer_data_sync(direct) key=customer_data_sync
                .> delete_archived_data exchange=delete_archived_data(direct) key=delete_archived_data
                .> download         exchange=download(direct) key=download
                .> process          exchange=process(direct) key=process
                .> query_upload     exchange=query_upload(direct) key=query_upload
                .> remove_expired   exchange=remove_expired(direct) key=remove_expired
                .> reporting        exchange=reporting(direct) key=reporting
                .> upload           exchange=upload(direct) key=upload

[tasks]
  . koku.metrics.collect_metrics
  . masu.celery.tasks.check_report_updates
  . masu.celery.tasks.delete_archived_data
  . masu.celery.tasks.query_and_upload_to_s3
  . masu.celery.tasks.remove_expired_data
  . masu.celery.tasks.sync_data_to_customer
  . masu.celery.tasks.upload_normalized_data
  . masu.celery.tasks.vacuum_schemas
  . masu.processor.tasks.get_report_files
  . masu.processor.tasks.refresh_materialized_views
  . masu.processor.tasks.remove_expired_data
  . masu.processor.tasks.summarize_reports
  . masu.processor.tasks.update_all_summary_tables
  . masu.processor.tasks.update_charge_info
  . masu.processor.tasks.update_summary_tables
  . masu.processor.tasks.vacuum_schema

[2020-02-12 18:35:56,980] INFO 3ef44759-af16-47e3-99cb-98050a6c3b8a Polling for
 schema_name: acct10001
 provider: Azure
 account (provider uuid): f4797201-562f-4eb8-8af2-8dff2422095a
[2020-02-12 18:35:57,070] INFO 3ef44759-af16-47e3-99cb-98050a6c3b8a Getting January 2020 report files for account (provider uuid): f4797201-562f-4eb8-8af2-8dff2422095a
[2020-02-12 18:35:57,163] INFO 3ef44759-af16-47e3-99cb-98050a6c3b8a Download queued - schema_name: acct10001, Task ID: 22d86f57-e0c0-42f5-9c47-d3decb203f17
[2020-02-12 18:35:57,170] INFO 3ef44759-af16-47e3-99cb-98050a6c3b8a Getting February 2020 report files for account (provider uuid): f4797201-562f-4eb8-8af2-8dff2422095a
[2020-02-12 18:35:57,235] INFO 3ef44759-af16-47e3-99cb-98050a6c3b8a Download queued - schema_name: acct10001, Task ID: 520e070b-b3a4-48a5-a942-2a9242dd894a
[2020-02-12 18:35:57,244: INFO/ForkPoolWorker-1] [masu.processor.tasks.get_report_files(962ab636-c107-4410-8dd5-d28fe6684684 via 3ef44759-af16-47e3-99cb-98050a6c3b8a)] Downloading report for:
 schema_name: acct10001
 provider: Azure
 account (provider uuid): f4797201-562f-4eb8-8af2-8dff2422095a
 report_month: January 2020
[2020-02-12 18:35:57,244: INFO/ForkPoolWorker-1] [masu.processor.tasks.get_report_files(962ab636-c107-4410-8dd5-d28fe6684684 via 3ef44759-af16-47e3-99cb-98050a6c3b8a)] Available disk space: 8416829440 bytes (26.099999999999994%)
[2020-02-12 18:35:59,243] INFO 3ef44759-af16-47e3-99cb-98050a6c3b8a Attempting to get Azure manifest for 2020-01-01 00:00:00...
[2020-02-12 18:35:59,246: INFO/ForkPoolWorker-1] Request URL: 'https://mysa1.blob.core.windows.net/output?prefix=REDACTED&restype=REDACTED&comp=REDACTED'
[2020-02-12 18:35:59,247: INFO/ForkPoolWorker-1] Request method: 'GET'
[2020-02-12 18:35:59,247: INFO/ForkPoolWorker-1] Request headers:
[2020-02-12 18:35:59,247: INFO/ForkPoolWorker-1]     'Accept': 'application/xml'
[2020-02-12 18:35:59,247: INFO/ForkPoolWorker-1]     'x-ms-version': 'REDACTED'
[2020-02-12 18:35:59,247: INFO/ForkPoolWorker-1]     'x-ms-date': 'REDACTED'
[2020-02-12 18:35:59,248: INFO/ForkPoolWorker-1]     'x-ms-client-request-id': '83734abe-4dc6-11ea-833f-0a580a800174'
[2020-02-12 18:35:59,248: INFO/ForkPoolWorker-1]     'User-Agent': 'azsdk-python-storage-blob/12.1.0 Python/3.6.9 (Linux-4.18.0-147.3.1.el8_1.x86_64-x86_64-with-redhat-7.7-Maipo)'
[2020-02-12 18:35:59,248: INFO/ForkPoolWorker-1]     'Authorization': 'REDACTED'
[2020-02-12 18:35:59,608: INFO/ForkPoolWorker-1] Response status: 200
[2020-02-12 18:35:59,611: INFO/ForkPoolWorker-1] Response headers:
[2020-02-12 18:35:59,611: INFO/ForkPoolWorker-1]     'Transfer-Encoding': 'chunked'
[2020-02-12 18:35:59,611: INFO/ForkPoolWorker-1]     'Content-Type': 'application/xml'
[2020-02-12 18:35:59,612: INFO/ForkPoolWorker-1]     'Server': 'Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0'
[2020-02-12 18:35:59,612: INFO/ForkPoolWorker-1]     'x-ms-request-id': 'REDACTED'
[2020-02-12 18:35:59,612: INFO/ForkPoolWorker-1]     'x-ms-client-request-id': '83734abe-4dc6-11ea-833f-0a580a800174'
[2020-02-12 18:35:59,612: INFO/ForkPoolWorker-1]     'x-ms-version': 'REDACTED'
[2020-02-12 18:35:59,612: INFO/ForkPoolWorker-1]     'Date': 'Wed, 12 Feb 2020 18:35:59 GMT'
[2020-02-12 18:35:59,629] INFO 3ef44759-af16-47e3-99cb-98050a6c3b8a Unable to find manifest. Error: No cost report found in container output for path cost/costreport/20200101-20200131.
[2020-02-12 18:35:59,651: INFO/ForkPoolWorker-1] [masu.processor.tasks.get_report_files(962ab636-c107-4410-8dd5-d28fe6684684 via 3ef44759-af16-47e3-99cb-98050a6c3b8a)] Reports to be processed:
 schema_name: acct10001
 provider: Azure
 provider_uuid: f4797201-562f-4eb8-8af2-8dff2422095a
[2020-02-12 18:35:59,666: INFO/ForkPoolWorker-1] [masu.processor.tasks.get_report_files(08f12ca2-2c3b-4e46-994f-a52efcfe8653 via 3ef44759-af16-47e3-99cb-98050a6c3b8a)] Downloading report for:
 schema_name: acct10001
 provider: Azure
 account (provider uuid): f4797201-562f-4eb8-8af2-8dff2422095a
 report_month: February 2020
[2020-02-12 18:35:59,667: INFO/ForkPoolWorker-1] [masu.processor.tasks.get_report_files(08f12ca2-2c3b-4e46-994f-a52efcfe8653 via 3ef44759-af16-47e3-99cb-98050a6c3b8a)] Available disk space: 8416673792 bytes (26.099999999999994%)
[2020-02-12 18:36:02,006] INFO 3ef44759-af16-47e3-99cb-98050a6c3b8a Attempting to get Azure manifest for 2020-02-01 00:00:00...
[2020-02-12 18:36:02,008: INFO/ForkPoolWorker-1] Request URL: 'https://mysa1.blob.core.windows.net/output?prefix=REDACTED&restype=REDACTED&comp=REDACTED'
[2020-02-12 18:36:02,008: INFO/ForkPoolWorker-1] Request method: 'GET'
[2020-02-12 18:36:02,008: INFO/ForkPoolWorker-1] Request headers:
[2020-02-12 18:36:02,008: INFO/ForkPoolWorker-1]     'Accept': 'application/xml'
[2020-02-12 18:36:02,009: INFO/ForkPoolWorker-1]     'x-ms-version': 'REDACTED'
[2020-02-12 18:36:02,009: INFO/ForkPoolWorker-1]     'x-ms-date': 'REDACTED'
[2020-02-12 18:36:02,009: INFO/ForkPoolWorker-1]     'x-ms-client-request-id': '8518cb1e-4dc6-11ea-833f-0a580a800174'
[2020-02-12 18:36:02,009: INFO/ForkPoolWorker-1]     'User-Agent': 'azsdk-python-storage-blob/12.1.0 Python/3.6.9 (Linux-4.18.0-147.3.1.el8_1.x86_64-x86_64-with-redhat-7.7-Maipo)'
[2020-02-12 18:36:02,009: INFO/ForkPoolWorker-1]     'Authorization': 'REDACTED'
[2020-02-12 18:36:02,189: INFO/ForkPoolWorker-1] Response status: 200
[2020-02-12 18:36:02,190: INFO/ForkPoolWorker-1] Response headers:
[2020-02-12 18:36:02,190: INFO/ForkPoolWorker-1]     'Transfer-Encoding': 'chunked'
[2020-02-12 18:36:02,190: INFO/ForkPoolWorker-1]     'Content-Type': 'application/xml'
[2020-02-12 18:36:02,190: INFO/ForkPoolWorker-1]     'Server': 'Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0'
[2020-02-12 18:36:02,190: INFO/ForkPoolWorker-1]     'x-ms-request-id': 'REDACTED'
[2020-02-12 18:36:02,190: INFO/ForkPoolWorker-1]     'x-ms-client-request-id': '8518cb1e-4dc6-11ea-833f-0a580a800174'
[2020-02-12 18:36:02,190: INFO/ForkPoolWorker-1]     'x-ms-version': 'REDACTED'
[2020-02-12 18:36:02,190: INFO/ForkPoolWorker-1]     'Date': 'Wed, 12 Feb 2020 18:36:01 GMT'
[2020-02-12 18:36:02,191] INFO 3ef44759-af16-47e3-99cb-98050a6c3b8a Unable to find manifest. Error: No cost report found in container output for path cost/costreport/20200201-20200229.
[2020-02-12 18:36:02,208: INFO/ForkPoolWorker-1] [masu.processor.tasks.get_report_files(08f12ca2-2c3b-4e46-994f-a52efcfe8653 via 3ef44759-af16-47e3-99cb-98050a6c3b8a)] Reports to be processed:
 schema_name: acct10001
 provider: Azure
 provider_uuid: f4797201-562f-4eb8-8af2-8dff2422095a

```
```

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -4,64 +4,63 @@ verify_ssl = true
...
+coveralls = ">=1.10"
+python-coveralls = ">=2.9"
+model-bakery = ">=1.1"
 koku-nise = "~=1.0"
-pre-commit = "*"
+pre-commit = ">=2.0"
 
 [requires]
-python_version = "3.6"
+python_version = ">=3.6"
```
Comment (myersCody) [inline Pipfile]: So, I get an error with this on mac:

```
cmyers-mac:koku cmyers$ pipenv --rm
Removing virtualenv (/Users/cmyers/.local/share/virtualenvs/koku-8P34cb70)…
cmyers-mac:koku cmyers$ pipenv install --dev --sequential
Traceback (most recent call last):
  File "/usr/local/Cellar/pipenv/2018.11.26_2/libexec/bin/pipenv", line 11, in <module>
    load_entry_point('pipenv==2018.11.26', 'console_scripts', 'pipenv')()
  File "/usr/local/Cellar/pipenv/2018.11.26_2/libexec/lib/python3.7/site-packages/pipenv/vendor/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/Cellar/pipenv/2018.11.26_2/libexec/lib/python3.7/site-packages/pipenv/vendor/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/local/Cellar/pipenv/2018.11.26_2/libexec/lib/python3.7/site-packages/pipenv/vendor/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/Cellar/pipenv/2018.11.26_2/libexec/lib/python3.7/site-packages/pipenv/vendor/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/Cellar/pipenv/2018.11.26_2/libexec/lib/python3.7/site-packages/pipenv/vendor/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/Cellar/pipenv/2018.11.26_2/libexec/lib/python3.7/site-packages/pipenv/vendor/click/decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
  File "/usr/local/Cellar/pipenv/2018.11.26_2/libexec/lib/python3.7/site-packages/pipenv/vendor/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/Cellar/pipenv/2018.11.26_2/libexec/lib/python3.7/site-packages/pipenv/vendor/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/local/Cellar/pipenv/2018.11.26_2/libexec/lib/python3.7/site-packages/pipenv/cli/command.py", line 254, in install
    editable_packages=state.installstate.editables,
  File "/usr/local/Cellar/pipenv/2018.11.26_2/libexec/lib/python3.7/site-packages/pipenv/core.py", line 1741, in do_install
    pypi_mirror=pypi_mirror,
  File "/usr/local/Cellar/pipenv/2018.11.26_2/libexec/lib/python3.7/site-packages/pipenv/core.py", line 574, in ensure_project
    pypi_mirror=pypi_mirror,
  File "/usr/local/Cellar/pipenv/2018.11.26_2/libexec/lib/python3.7/site-packages/pipenv/core.py", line 494, in ensure_virtualenv
    python = ensure_python(three=three, python=python)
  File "/usr/local/Cellar/pipenv/2018.11.26_2/libexec/lib/python3.7/site-packages/pipenv/core.py", line 397, in ensure_python
    path_to_python = find_a_system_python(python)
  File "/usr/local/Cellar/pipenv/2018.11.26_2/libexec/lib/python3.7/site-packages/pipenv/core.py", line 360, in find_a_system_python
    python_entry = finder.find_python_version(line)
  File "/usr/local/Cellar/pipenv/2018.11.26_2/libexec/lib/python3.7/site-packages/pipenv/vendor/pythonfinder/pythonfinder.py", line 95, in find_python_version
    version_dict = PythonVersion.parse(major)
  File "/usr/local/Cellar/pipenv/2018.11.26_2/libexec/lib/python3.7/site-packages/pipenv/vendor/pythonfinder/models/python.py", line 359, in parse
    version_dict = parse_python_version(str(version))
  File "/usr/local/Cellar/pipenv/2018.11.26_2/libexec/lib/python3.7/site-packages/pipenv/vendor/pythonfinder/utils.py", line 86, in parse_python_version
    raise InvalidPythonVersion("%s is not a python version" % version_str)
pipenv.vendor.pythonfinder.exceptions.InvalidPythonVersion: >=3.6 is not a python version
```

Comment (blentz) [reply Pipfile]: That's fine. It causes me errors on Fedora, because Fedora is on Python 3.7. But I don't mind working around it for a while longer.

Comment (myersCody) [reply Pipfile]: Update: https://github.com/pypa/pipenv/issues/1050
I think this is just a common pipenv thing, seems like devs want the ability to do it, but its a no go from the pipenv developers.

Comment (myersCody) [reply Pipfile]: @blentz, totally my default python version for mac is 3.7, but the pipenv should handle the python versioning mess for you. Are you using pipenv on fedora?

**Pull request merged**